### ISSUE TITLE
Incorrect android system resource assigned in Robolectric 4.11

### ISSUE THREAD

**Issue by charlesng** [robolectric/robolectric#8578]
### Description

I have a custom android view with custom attribute to accept a drawable resource. I wrote a test to verify whether this custom attribute work as expected, the data i used in the test case is android drawable resource ,not from the project. However, i got error after i upgrade to Robolectric 4.11 , i didn't get error in Robolectric 4.10.3. 

### Steps to Reproduce

Here is a reproducible test case for a simple imageview  in a hello world project

```kt
@RunWith(AndroidJUnit4::class)
class ExampleUnitTest {
    @Test
    fun should_set_correct_drawable_for_image() {
        // given
        val activityController = Robolectric.buildActivity(Activity::class.java)
        val activity = activityController.get()
        val attrSet = Robolectric.buildAttributeSet()
            .addAttribute(android.R.attr.src, "@android:drawable/btn_star")
            .build()

        // when
        val imageView = ImageView(activity, attrSet)

        // then
        val resId = shadowOf(imageView.drawable).createdFromResId
        val resName = ApplicationProvider.getApplicationContext<Application>()
            .resources.getResourceName(resId)
        assertEquals("android:drawable/btn_star", resName) // the value become 'android:drawable/ic_delete' in 4.11
        assertEquals(android.R.drawable.btn_star, resId) // the value become 'android.R.drawable.ic_delete' in 4.11
    }
}
```

You see the test passed in 4.10.3 but failed in 4.11 and  the resource name become `android:drawable/ic_delete` if i upgrade to Robolectric 4.11

### Robolectric & Android Version
- Robolectric 4.11
- AGP 8.1.2
- compile sdk 34 

### Link to a public git repo demonstrating the problem:

- N/A

**Comment by hoisie** [comment]
If it's not too much trouble, could you post a sample app on GitHub demonstrating the problem?

**Comment by hoisie** [comment]
Actually nevermind, I believe I am able to repro the example with just that test case, thanks 👍

**Comment by hoisie** [comment]
I am running a git bisect to look into this issue.

Note that we encourage people to use LayoutInflater to inflate views instead of using AttributeSetBuilder. AttributeSetBuilder is not maintained very much, it kind of predated binary resources.

**Comment by hoisie** [comment]
After a git bisect, this issue started when Android U was added to the default SDK list: 54087220d5373273c830b940e40826c332e5e864

**Comment by hoisie** [comment]
I believe I have a fix for this issue, see the attached commit.

**Comment by hoisie** [comment]
This should be fixed in the 4.11.1 release. Let me know if there are any other resource-related issues. Android U uses a different encoding scheme (sparse encoding) for some system resource tables, and there may still be issues lingering related to that.
